OpenSim/Common/Component.h

```diff
@@ -706,34 +714,24 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
 
 
     /** @name Component Connector Access methods
-        Access Connectors of this component in a generic way and also by name.
-    */
+        Access Connectors of this component by name. */
     //@{ 
-    
-    /** Get the number of Connectors and then access a Connector by index.
-        For example:
-        @code
-        for (int i = 0; i < myComp.getNumConnectors(); ++i){
-        const AbstractConnector& connector = myComp.getConnector(i);
-        // check status: e.g. is it connected?
-        ...
-        AbstractConnector& connector = myComp.updConnector(i);
-        // change the status: e.g. disconnect or change/define connectee;
-        }
-        @endcode
-        @see getNumConnectors()
-        @see getConnector(int i);
-     */
+    /** Get the number of Connectors in this Component. */
     int getNumConnectors() const {
-        return getProperty_connectors().size();
+        return int(_connectorsTable.size());
     }
 
-    /** Collect and return the names of the connectors in this
-     * component. The names are in the same order as the connectors. */
+    /** Collect and return the names of the connectors in this component. You
+     * can use this to iterate through the connectors:
+     * @code
+     * for (std::string name : comp.getConnectorNames()) {
+     *     const AbstractConnector& conn = getConnector(name);
+     * }
+     * @endcode */
     std::vector<std::string> getConnectorNames() {
-        std::vector<std::string> names(getNumConnectors());
-        for (int i = 0; i < getNumConnectors(); ++i) {
-            names[i] = get_connectors(i).getName(); 
+        std::vector<std::string> names;
+        for (const auto& it : _connectorsTable) {
+            names.push_back(it.first);
```

Comment:
> now it is the order maintained by the map

Correct.

> so this might be OK.

I haven't thought of a problem with this yet. But if there is, we can change the data structure from a map to a vector.